Add upgrade note for Database interface change in 1.6 (#10372)
hashicorp/vault: website/pages/docs/upgrading/upgrade-to-1.6.0.mdx

type oriented configuration API.  Secondaries will receive the 
modifications via replication. 

## Database Engine Interface Upgrade

The Database Engine has changed the underlying interface between Vault and each database
implementation. This change allows use of [password policies](/docs/concepts/password-policies)
within the Database engine. The API for the Database Engine has not changed, only the underlying
interface between Vault and the database plugins. All built-in database plugins (as well as the
[Oracle](https://github.com/hashicorp/vault-plugin-database-oracle) plugin) have been upgraded to
the new interface so no user actions are needed. Vault will continue to recognize existing custom

> database plugins but that functionality should be considered deprecated and may be removed in a

Reviewer: ```suggestion
database plugins but the old interface should be considered deprecated and may be removed in a
```
